File: intro_html/calculadora.py
from flask import Flask, jsonify, request
from flask_cors import CORS

app = Flask(__name__)
CORS(app)

@app.route("/")
def inicio():
    return "Hola bonito mundo",200

@app.route("/suma", methods=["POST"])
def suma():
    fNumber = int(request.form["first"])
    sNumber = int(request.form["second"])
    resultado = fNumber + sNumber
    return str(resultado)

@app.route("/resta", methods=["POST"])
def resta():
    fNumber = int(request.form["first"])
    sNumber = int(request.form["second"])
    resultado = fNumber - sNumber
    return str(resultado)

@app.route("/multi", methods=["POST"])
def multiplicacion():
    fNumber = int(request.form["first"])
    sNumber = int(request.form["second"])
    resultado = fNumber * sNumber
    return str(resultado)

@app.route("/div", methods=["POST"])
def division():
    fNumber = int(request.form["first"])
    sNumber = int(request.form["second"])
    resultado = fNumber / sNumber
    return str(resultado)

@app.route("/login", methods=["POST"])
def inicioSesion():
    usr = request.form["username"]
    ps = request.form["password"]
    if usr == "admin" and ps == "1234":
        return "Bienvenido admin"
    else:
        return "Usuario o contraseña incorrectos"

File: intro_html/calculadora.js
$(document).ready(function() {
    $("#suma").click(function() {
        let first = 0;
        let second = 0;
        first = Number($("#fnumber").val());
        second = Number($("#snumber").val());
        valores = {"first": first, "second": second}
        $.post("http://127.0.0.1:5000/suma", valores, function(data, status){
            console.log("Estado: " + status);
            $("#result").text(data);
        });
    });
    $("#resta").click(function() {
        let first = 0;
        let second = 0;
        first = Number($("#fnumber").val());
        second = Number($("#snumber").val());
        valores = {"first": first, "second": second}
        $.post("http://127.0.0.1:5000/resta", valores, function(data, status){
            console.log("Estado: " + status);
            $("#result").text(data);
        });
    });
    $("#multi").click(function() {
        let first = 0;
        let second = 0;
        first = Number($("#fnumber").val());
        second = Number($("#snumber").val());
        valores = {"first": first, "second": second}
        $.post("http://127.0.0.1:5000/multi", valores, function(data, status){
            console.log("Estado: " + status);
            $("#result").text(data);
        });
    });
    $("#div").click(function() {
        let first = 0;
        let second = 0;
        first = Number($("#fnumber").val());
        second = Number($("#snumber").val());
        valores = {"first": first, "second": second}
        $.post("http://127.0.0.1:5000/div", valores, function(data, status){
            console.log("Estado: " + status);
            $("#result").text(data);
        });
    });
    $("#login").click(function() {
        let username = $("#username").val();
        let password = $("#password").val();
        valores = {"username": username, "password": password};
        $.post("http://127.0.0.1:5000/login", valores, function(data, status){
            alert("Inicio de sesión recibido: " + status);
            $("#resultLogin").text(data);
        });
    });
});